lib/common/attrs.c

```diff
@@ -22,6 +22,18 @@
 
 /*!
  * \internal
+ * \brief Get the node name that should be used to set node attributes
+ *
+ * If given NULL, "auto", or "localhost" as an argument, check the environment
+ * to detect the node name that should be used to set node attributes. (The
+ * caller might not know the correct name, for example if the target is part of
+ * a bundle with container-attribute-target set to "host".)
+ *
+ * \param[in] name  NULL, "auto" or "localhost" to check environment variables,
+ *                  or anything else to return NULL
+ *
+ * \return Node name that should be used for node attributes based on the
+ *         environment if known, otherwise NULL
```

Comment:
> If the `name` argument is `"auto"` (or `"localhost"`), `host_physical == NULL && !pcmk__str_eq(target, "host", pcmk__str_casei)`, and `getenv(LRM_TARGET_ENV) == NULL`, then we return `"auto"` (or `"localhost"`). We should probably update the function to return `NULL` in those edge cases -- maybe by setting `name = NULL` right after the initializations.

This is only expected to be called for crm_attribute within a resource agent, in which case LRM_TARGET_ENV will be set, so it hasn't been a problem in practice. But that definitely makes sense and it's probably causing some edge case issues.

> I question whether we need three arguments that do the same thing in an internal function anyway.

I believe resource agents are allowed to specify auto or localhost when setting node attributes. Probably wasn't the best idea ...

Reply:
pacemaker-attrd will map "localhost" to the local node name, so that should stay as-is. "auto" is questionable whether that should be allowed but I'd rather not mess with it at this point

Reply:
I only looked into it because of the new `\return` string, which turned out to be incorrect for a couple of rare edge cases. One option would be to leave the code alone and update the `\return` string.